modules/internal/ChapelArray.chpl

```diff
@@ -2293,7 +2295,17 @@ module ChapelArray {
              "  Actual domain is: ", this.domain);
     }
 
-    pragma "no doc"
+    /* Return an array declared over a new domain, provided that the new
+       domain is of the same rank and size as the original.
+
+       For example:
+
+       .. code-block:: chapel
+
+          var A: [{1..10}] int;
```

Comment:
I'd remove these curly brackets.  I don't think we want to propagate a notion that this is required for Chapel (nor the preferred style).